[FIXED JENKINS-32852] Fixed possible ArrayIndexOutOfBoundsException

pjanouse opened the pull request:
Fixed possible ArrayIndexOutOfBoundsException in hudson.model.Fingerprint.RangeSet.fromString(...)

daniel-beck commented:
Appears to introduce a :bug: as it doesn't check `skipError` as the existing `catch` block does.

pjanouse commented:
All right, my bad, I will fix it.
Catching it in addition to NFE isn't enough because we need to reject invalid values like "1-3-4" as well, so strict check for available tokens would be necessary I think.

olivergondza commented:
LGTM

oleg-nenashev commented on core/src/main/java/hudson/model/Fingerprint.java:
```diff
@@ -694,7 +694,14 @@ public static RangeSet fromString(String list, boolean skipError) {
+                        }
+                        else {
+                            if (!skipError)
+                                throw new IllegalArgumentException("Unable to parse " + list);
```
NIT: missing brackets. It would be also useful to explain the formatting requirements in the exception. E.g. "expected string with format XX-YY" or something like that. Maybe Int parsing errors should be caught & processed as well

pjanouse replied:
NFE is caught later, exception messages improved.

oleg-nenashev commented on core/src/main/java/hudson/model/Fingerprint.java:
```diff
@@ -694,7 +694,14 @@ public static RangeSet fromString(String list, boolean skipError) {
+                        if (tokens.length == 2) {
+                            rs.ranges.add(new Range(Integer.parseInt(tokens[0]), Integer.parseInt(tokens[1]) + 1));
+                        }
+                        else {
```
NIT: line break before else

oleg-nenashev commented:
LGTM in general, added a minor comment regarding the possible improvement

pjanouse committed:
[JENKINS-32852] Fixed possible ArrayIndexOutOfBoundsException in RangeSet.fromString()

Fixed possible ArrayIndexOutOfBoundsException in
hudson.model.Fingerprint.RangeSet.fromString(...)
